### ISSUE TITLE
Can lightgbm extract features automatically

### ISSUE THREAD

**Issue by zhangyu345293721** [microsoft/LightGBM#4339]

**Comment by jameslamb** [comment]
Thanks for using LightGBM!

LightGBM uses tree learners, and the tree construction process is a form of feature selection...only features which have splits that improve the training loss will be chosen.

LightGBM's `Dataset` object also does some preprocessing that you can think of as feature engineering. For example, through its special encoding of categorical variables and its construction of compound features to deal with sparse data.

For details, please see the following:

* https://lightgbm.readthedocs.io/en/latest/Features.html
* https://lightgbm.readthedocs.io/en/latest/Advanced-Topics.html
* https://papers.nips.cc/paper/2017/file/6449f44a102fde848669bdd9eb6b76fa-Paper.pdf

<hr>

If this did not answer your question, please ask a more specific question and we'd be happy to re-open this and help.
